salt/state.py

```diff
@@ -2507,22 +2508,16 @@ def _get_envs(self):
             envs.extend([x for x in list(self.opts['file_roots'])
                          if x not in envs])
         env_order = self.opts.get('env_order', [])
+        # Remove duplicates
+        env_order = list(collections.OrderedDict.fromkeys(env_order))
```

Comment:
OK, but you can still iterate over a list or use a list comprehension and not incur all that additional overhead.